slackclient/server.py

```diff
@@ -347,8 +347,16 @@ def api_call(self, token, request="?", timeout=None, **kwargs):
             See here for more information on responses: https://api.slack.com/web
         """
         response = self.api_requester.do(token, request, kwargs, timeout=timeout)
+
+        if response.status_code == 429:
```

Comment:
Instead of infinitely recursing could you move the retry logic into a decorator? This decorator should also use a default maximum 3 of retries.